pkg/scheduler/algorithm_factory.go

```diff
@@ -382,16 +383,36 @@ func RegisterCustomPriorityFunction(policy schedulerapi.PriorityPolicy, args *pl
 	// generate the priority function, if a custom priority is requested
 	if policy.Argument != nil {
 		if policy.Argument.ServiceAntiAffinity != nil {
-			pcf = &PriorityConfigFactory{
-				MapReduceFunction: func(args PluginFactoryArgs) (priorities.PriorityMapFunction, priorities.PriorityReduceFunction) {
-					return priorities.NewServiceAntiAffinityPriority(
-						args.PodLister,
-						args.ServiceLister,
-						policy.Argument.ServiceAntiAffinity.Label,
-					)
-				},
-				Weight: policy.Weight,
+			// We use the ServiceAffinity plugin name for all ServiceAffinity custom priorities.
+			// It may get called multiple times but we essentially only register one instance of
+			// ServiceAffinity priority.
+			// This name is then used to find the registered plugin and run the plugin instead of the priority.
+			name = serviceaffinity.Name
+
+			if args.ServiceAffinityArgs != nil {
```

Comment:
Seems that this should be `if args.ServiceAffinityArgs == nil`

Reply:
Thanks for pointing this out!